guides/common/modules/proc_installing-puppet-agent-during-host-provisioning.adoc

```diff
@@ -1,30 +1,63 @@
+For more information, see {ContentManagementDocURL}Importing_Content_content-management[Importing Content] in _{ContentManagementDocTitle}_.
+* You created an activation key that enables the Puppet agent repository for hosts.
+For more information, see {ContentManagementDocURL}Managing_Activation_Keys_content-management[Managing Activation Keys] in _{ContentManagementDocTitle}_.
+* You assigned the activation key to a host group.
+endif::[]
 
 .Procedure
 . Navigate to *Hosts > Provisioning Templates*.
 . Select a provisioning template depending on your host provisioning method.
 For more information, see {ProvisioningDocURL}types-of-provisioning-templates_provisioning[Types of Provisioning Templates] in _{ProvisioningDocTitle}_.
+ifndef::katello,orcharhino,satellite[]
+. Ensure the `puppetlabs_repo` snippet is included as follows:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+<%= snippet 'puppetlabs_repo' %>
+----
+endif::[]
 . Ensure the `puppet_setup` snippet is included as follows:
 +
 [options="nowrap", subs="+quotes,verbatim,attributes"]
 ----
 <%= snippet 'puppet_setup' %>
 ----
-. Enable Puppet using a host parameter for a single host or host group:
+ifndef::katello,orcharhino,satellite[]
+. Enable the Puppet agent from the official Puppet repository using one of the following host parameters in global parameters, a host group, or for a single host:
+
+* Add a host parameter named `enable-puppetlabs-repo` for the latest stable Puppet release in the unversioned repository.
+* Add a host parameter named `enable-official-puppet7-repo` for the Puppet 7 release repository.
+* Add a host parameter named `enable-puppetlabs-puppet6-repo` for the Puppet 6 release repository.
+
++
+Select the `boolean` type and set the value to `true`.
+endif::[]
+ifdef::katello,orcharhino,satellite[]
+. Enable the Puppet agent using a host parameter in global parameters, a host group, or for a single host.
 +
-Add a host parameter named `enable-puppet7` for Puppet 7
-ifndef::satellite[]
-or `enable-puppet6` for Puppet 6
 endif::[]
-as type boolean set to `true`.
-. Optional: To install the Puppet agent directly from https://yum.puppet.com/[yum.puppet.com], add a host parameter named `enable-puppetlabs-puppet7-repo` for Puppet 7
-ifndef::satellite[]
-or `enable-puppetlabs-puppet6-repo` for Puppet 6
+ifdef::katello,orcharhino[]
+Add a host parameter named `enable-puppet7` for Puppet 7 or `enable-puppet6` for Puppet 6, select the `boolean` type and set the value to `true`.
 endif::[]
-as type boolean set to `true`.
-ifndef::katello[]
-Only use this if you don't provide Puppet agent to the host using its activation key.
+ifdef::satellite[]
+Add a host parameter named `enable-puppet7` for Puppet 7, select the `boolean` type and set the value to `true`.
 endif::[]
+ifdef::katello,orcharhino,satellite[]
+. When you create the host, assign the host group that contains the appropriate activation key.
```

Comment:
```suggestion
. Ensure your host has access to the Puppet agent from {ProjectServer} by using an appropriate activation key.
```

Reply:
During provisioning (Create Host), the user cannot assign an activation key directly, but only indirectly using a host group instead.

Reply:
If I go to hosts > create hosts; create a host; select a host group, then I can still overwrite the AK as host parameter or add one.

Reply:
If you mean the `kt_activation_keys` parameter, it seems that it's only interpreted like this in the Global Registration template and only when defined on the hostgroup level.
Or what parameters exactly do you mean?

Reply:
Oh, it's in a snippet. Never mind!